### PR TITLE
Add information on internship programs

### PR DESCRIPTION
**Pull request by brianwarner**
Adds information on internships that are available to projects which meet the mentorship
requirements. Currently links to the TAC minutes approving the initial round, but should be updated
each year to reflect the current process.

Signed-off-by: Brian Warner <brian@bdwarner.com>

### CODE REVIEW

**Review by dthaler — COMMENTED** [review]
Text needs to be added to https://github.com/confidential-computing/governance/blob/master/project-progression-policy.md#benefits-of-being-a-recognized-consortium-project

File: diversity-and-inclusion-policies.md [hunk 1]
```diff
@@ -48,3 +48,5 @@ https://github.com/torvalds/linux/blob/a5f526ecb075a08c4a082355020166c7fe13ae27/
 https://docs.microsoft.com/en-us/style-guide/bias-free-communication
 https://developers.google.com/style/inclusive-documentation
 
+### Internships
+The CCC TAC and Outreach committee have [committed to support interns](./project-mentors.md) for CCC projects that meet the mentor requirements.
```
Comment (dthaler) [inline diversity-and-inclusion-policies.md:52]: This mentions "mentor requiremements" but doesn't say what they are.  https://www.outreachy.org/mentor

Also per Outreach meeting 10/20/2021 I think we can drop "Outreach committee" and just say "TAC"

File: project-mentors.md [hunk 2]
```diff
@@ -43,3 +43,7 @@ The following projects are in the middle of the contribution process and have te
 
 * **[Occlum](https://occlum.io)**: [Zongmin Gu](https://github.com/guzongmin)
 * **[SGX SDK for Linux](https://github.com/intel/linux-sgx)**: [Simon Johnson](https://github.com/spjohnso)
+
+## Internships
+The [TAC and Outreach committe have committed](https://lists.confidentialcomputing.io/g/main/files/TAC/Meetings/2021/09-Sept/TAC%20Minutes%202021-09-09.pdf) to fund one intern per CCC project that meets the mentorship guidelines.
```
Comment (dthaler) [inline project-mentors.md:48]:
```suggestion
The [TAC and Outreach committees have committed](https://lists.confidentialcomputing.io/g/main/files/TAC/Meetings/2021/09-Sept/TAC%20Minutes%202021-09-09.pdf) to fund one intern per CCC project that meets the mentorship guidelines.
```
Also, This mentions "mentor requiremements" but doesn't say what they are. https://www.outreachy.org/mentor

**Review by dthaler — APPROVED** [review]

**Review by dcmiddle — APPROVED** [review]
Looks good. Please squash on merge.

**Comment by dthaler** [comment]
Merging per rule 4 of https://github.com/confidential-computing/governance/blob/master/tac-github-process.md

**Pull request merged**